From 9722bc2fde8f8015c9ae48498ad54e63ac2cca86 Mon Sep 17 00:00:00 2001
From: Sean O'Keeffe <seanokeeffe797@gmail.com>
Date: Fri, 24 Nov 2017 14:02:58 +0000
Subject: [PATCH 2/2] backport MissingValueError

---
 nailgun/entity_mixins.py | 27 +++++++++++++++------------
 1 file changed, 15 insertions(+), 12 deletions(-)

diff --git a/nailgun/entity_mixins.py b/nailgun/entity_mixins.py
index ab831f9..342b3c0 100644
--- a/nailgun/entity_mixins.py
+++ b/nailgun/entity_mixins.py
@@ -242,19 +242,22 @@ def _get_entity_id(field_name, attrs):
 
     """
     field_name_id = field_name + '_id'
-    if field_name in attrs:
-        if attrs[field_name] is None:
-            return None
+    error_msg = """ Cannot find a value for the "{0}" field. Searched
+                    for keys named {1}, but available keys are {2}.
+                """.format(field_name, (field_name, field_name_id),
+                           attrs.keys())
+    try:
+        if field_name in attrs:
+            if attrs[field_name] is None:
+                return None
+            elif 'id' in attrs[field_name]:
+                return attrs[field_name]['id']
+        if field_name_id in attrs:
+            return attrs[field_name_id]
         else:
-            return attrs[field_name]['id']
-    elif field_name_id in attrs:
-        return attrs[field_name_id]
-    else:
-        raise MissingValueError(
-            'Cannot find a value for the "{0}" field. Searched for keys named '
-            '{1}, but available keys are {2}.'
-            .format(field_name, (field_name, field_name_id), attrs.keys())
-        )
+            raise MissingValueError(error_msg)
+    except KeyError:
+        raise MissingValueError(error_msg)
 
 
 def _get_entity_ids(field_name, attrs):
-- 
2.13.6
