From 332da8d8cebff35aaae00dac9ccdd455157f6d13 Mon Sep 17 00:00:00 2001
From: Eric D Helms <eric.d.helms@gmail.com>
Date: Wed, 5 Oct 2016 11:47:18 -0400
Subject: [PATCH] Allow content view versions to be searched (#322)

---
 nailgun/entities.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/nailgun/entities.py b/nailgun/entities.py
index f2a3964..768bf4d 100644
--- a/nailgun/entities.py
+++ b/nailgun/entities.py
@@ -1457,7 +1457,11 @@ def __init__(self, server_config=None, **kwargs):
         super(ContentUpload, self).__init__(server_config, **kwargs)
 
 
-class ContentViewVersion(Entity, EntityReadMixin, EntityDeleteMixin):
+class ContentViewVersion(
+        Entity,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin):
     """A representation of a Content View Version non-entity."""
 
     def __init__(self, server_config=None, **kwargs):
