From 4914ef7a3c6f660aa23c310038cdab9abfd1814f Mon Sep 17 00:00:00 2001
From: Sean O'Keeffe <seanokeeffe797@gmail.com>
Date: Thu, 30 Nov 2017 17:54:53 +0000
Subject: [PATCH] search subs

---
 nailgun/entity_mixins.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/nailgun/entity_mixins.py b/nailgun/entity_mixins.py
index ab831f9..558f79b 100644
--- a/nailgun/entity_mixins.py
+++ b/nailgun/entity_mixins.py
@@ -281,7 +281,10 @@ def _get_entity_ids(field_name, attrs):
     elif field_name in attrs:
         return [entity['id'] for entity in attrs[field_name]]
     elif plural_field_name in attrs:
-        return [entity['id'] for entity in attrs[plural_field_name]]
+        try:
+            return [entity['id'] for entity in attrs[plural_field_name]]
+        except KeyError:
+            return [entity['host_id'] for entity in attrs[plural_field_name]]
     else:
         raise MissingValueError(
             'Cannot find a value for the "{0}" field. Searched for keys named '
-- 
2.13.6
